Claroty/claroty-xdome/ingest/parser.yml

```
@@ -0,0 +1,80 @@
+      name: date.parse
+      properties:
+        input_field: "{{parsed_event.message.event_timestamp}}"
+        output_field: date
+
+  - name: set_ecs_fields
+
+stages:
+  set_ecs_fields:
+    actions:
+      - set:
+          "@timestamp": "{{parsed_timestamp.date}}"
+
+      - set:
+          event.kind: "alert"
+
+      - set:
+          event.category: ["network"]
+          event.type: ["connection"]
+        filter: "{{parsed_event.message.username == None}}"
+
+      - set:
+          event.category: ["authentication"]
+          event.type: ["info"]
+        filter: "{{parsed_event.message.username != None}}"
+
+      - set:
+          event.severity: "{{parsed_event.message.Severity}}"
+          observer.vendor: "{{parsed_event.message.DeviceVendor}}"
+          observer.product: "{{parsed_event.message.DeviceProduct}}"
+          observer.version: "{{parsed_event.message.DeviceVersion}}"
+
+      - set:
+          source.ip: "{{parsed_event.message.src_ip}}"
+          source.mac: "{{parsed_event.message.src_mac}}"
+          source.port: "{{parsed_event.message.src_port}}"
+          user.name: "{{parsed_event.message.username}}"
+          destination.ip: "{{parsed_event.message.dst_ip}}"
+          destination.mac: "{{parsed_event.message.dst_mac}}"
+          destination.port: "{{parsed_event.message.dst_port}}"
+          destination.domain: "{{parsed_event.message.domain}}"
+          network.protocol: "{{parsed_event.message.protocol}}"
+          event.reason: "{{parsed_event.message.event_description}}"
+          event.code: "{{parsed_event.message.event_type}}"
+          host.mac: "{{parsed_event.message.device_mac_list}}"
+          host.ip: "{{parsed_event.message.device_ip_list}}"
+          device.id: "{{parsed_event.message.device_asset_id}}"
+          device.manufacturer: "{{parsed_event.message.device_manufacturer}}"
+          device.model.name: "{{parsed_event.message.device_model}}"
+          host.type: "{{parsed_event.message.device_type}}"
+          host.os.full: "{{parsed_event.message.device_os}}"
+
+      - set:
+          tls.server.x509.subject.common_name: "{{parsed_event.message.certificate_CN}}"
+          tls.server.x509.subject.country: "{{parsed_event.message.certificate_C}}"
+          tls.server.x509.subject.locality: "{{parsed_event.message.certificate_L}}"
+          tls.server.x509.subject.organization: "{{parsed_event.message.certificate_O}}"
+          tls.server.x509.subject.organizational_unit: "{{parsed_event.message.certificate_OU}}"
+          tls.server.x509.subject.state_or_province: "{{parsed_event.message.certificate_ST}}"
+          tls.server.x509.not_after: "{{parsed_event.message.certificate_expiry_date}}"
+          tls.server.x509.not_before: "{{parsed_event.message.certificate_start_date}}"
+          tls.server.hash.sha1: "{{parsed_event.message.certificate_thumbprint}}"
+        filter: "{{parsed_event.message.certificate_type in ('server', 'client') }}"
```

Lower the value of certificate type. Otherwise, the fields are not set in the generated events.
```suggestion
        filter: "{{parsed_event.message.certificate_type != null and parsed_event.message.certificate_type.lower() in ('server', 'client') }}"
```